wp-content/mu-plugins/pantheon/pantheon-updates.php

```diff
@@ -3,39 +3,64 @@
 if( ! defined('WP_AUTO_UPDATE_CORE')) {
 	define( 'WP_AUTO_UPDATE_CORE', false );
 }
-remove_action( 'wp_maybe_auto_update', 'wp_maybe_auto_update' );
 
-// Remove the default WordPress core update nag if on Pantheon
-if( isset( $_ENV['PANTHEON_ENVIRONMENT'] ) ){
-    add_action('admin_menu','_pantheon_hide_update_nag');
+// Short circuit WordPress checking WordPress.org for updates by returning zero updates and current time as last checked time
+function _pantheon_disable_wp_updates() {
+	// include an unmodified $wp_version
+	include( ABSPATH . WPINC . '/version.php' );
+
+	return (object) array(
+		'updates' => array(),
+		'version_checked' => $wp_version,
+		'last_checked' => time(),
+	);
+
 }
 
 function _pantheon_hide_update_nag() {
 	remove_action( 'admin_notices', 'update_nag', 3 );
+	remove_action( 'network_admin_notices', 'update_nag', 3 );
+}
+
+// Remove the default WordPress core update nag if on Pantheon
+if( isset( $_ENV['PANTHEON_ENVIRONMENT'] ) ){
+	add_filter( 'pre_site_transient_update_core', '_pantheon_disable_wp_updates' );
+    add_action('admin_menu','_pantheon_hide_update_nag');
 }
 
-// Get the latest WordPress version
+// Get the latest WordPress version tagged in the Pantheon upstream on GitHub
 function _pantheon_get_latest_wordpress_version() {
-	$core_updates = get_core_updates( array('dismissed' => false) );
+	$pantheon_latest_wp_version = get_transient( 'pantheon_latest_wp_version' );
+
+	if( false === $pantheon_latest_wp_version ){
+		$github_api_pantheon_upstream_tags_url = "https://api.github.com/repos/pantheon-systems/WordPress/git/refs/tags";
+		$response = wp_remote_get( $github_api_pantheon_upstream_tags_url );
+
+		if( ! is_wp_error( $response ) && 200 === $response['response']['code'] ){
```

Comment:
Also wanted to note if we receive anything but a `200` response from GitHub the current version of WordPress the site is using is returned. This will allow things to function successfully even if rate limits are hit or GitHub returns an error.

In that scenario it would still be about an hour for the rate limit to reset and the updated value to show the nag in the dashboard. The hour delay (for the transient or rate limit reset) only apply to the WordPress dashboard nag. The prompt to update on the Pantheon dashboard will still remain.

Reply:
"I don't know" is a good fallback strategy here.